arrow/arrow.py

```diff
@@ -1264,7 +1265,11 @@ def gather_timeframes(_delta: float, _frame: TimeFrameLiteral) -> float:
                     if _frame in granularity:
                         value = sign * _delta / self._SECS_MAP[_frame]
                         _delta %= self._SECS_MAP[_frame]
-                        if trunc(abs(value)) != 1:
+
+                        # If user chooses dynamic and the display value is 0 don't subtract
+                        if dynamic and trunc(abs(value)) == 0:
+                            pass
+                        elif trunc(abs(value)) != 1:
```

Comment:
Would it make any significant difference to save the value of `trunc(abs(value))` rather than calculating it twice?

Reply:
Decent catch.

Profiling the code, you'd need to run about 10k of the trunc(abs()) calls to even come close to seeing a 1ms difference.